src/dxtbx/nexus/nxmx.py

```diff
@@ -76,9 +87,12 @@ def find_classes(
     Returns:
         A list of matching nodes for each of the specified NX_class types.
     """
-    results = {nx_class: [] for nx_class in nx_classes}
+    results: Dict[Optional[str], List[h5py.Group]] = {
```

Comment:
Picking one at random:
if we added the future import everywhere this could become
```suggestion
    results: dict[Optional[str], list[h5py.Group]] = {
```
following what we did in https://github.com/dials/data/pull/318

I think that'd be good practice as this is going to be the future anyway, so people don't see (copy around) outdated coding conventions.

Reply:
Interesting! I knew that it came in with Python 3.9, but obviously hadn't joined the dots that as long as you don't try to evaluate the annotation in-interpreter it of course works fine. Yes, I think is better to go to early.